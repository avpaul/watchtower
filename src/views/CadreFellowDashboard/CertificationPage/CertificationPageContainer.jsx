import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import CertificationPage from './CertificationPage';
import { getCertification as getCertificationAction } from '../../../redux/actionCreators/getCertificationActions';

export const mapStateToProps = ({ getCertification }) => ({ getCertification });

export default connect(
  mapStateToProps,
  { getCertificationAction }
)(withRouter(CertificationPage));
