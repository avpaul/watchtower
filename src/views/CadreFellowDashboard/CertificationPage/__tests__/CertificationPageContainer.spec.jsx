import { mapStateToProps } from '../CertificationPageContainer';

describe('Test the certification container', () => {
  it('map state to props should return props value', () => {
    expect(
      mapStateToProps(
        {
          getCertification: {
            loading: false,
            data: {
              1: {}
            }
          },
          cadreVacancies: {
            data: {
              certificationVacancies: {}
            }
          }
        },
        { location: { pathname: '/dashboard/certifications/1' } }
      )
    ).toMatchSnapshot();
  });
  it('map state to props should return empty object on single certification when certification data is empty', () => {
    expect(
      mapStateToProps(
        {
          getCertification: {
            loading: false,
            data: {}
          },
          cadreVacancies: {
            data: {
              certificationVacancies: {}
            }
          }
        },
        { location: { pathname: '/dashboard/certifications/1' } }
      )
    ).toMatchSnapshot();
  });
});
