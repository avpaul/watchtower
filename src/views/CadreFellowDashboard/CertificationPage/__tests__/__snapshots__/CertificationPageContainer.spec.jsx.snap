// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the certification container map state to props should return empty object on single certification when certification data is empty 1`] = `
Object {
  "certificationApplication": undefined,
  "certificationVacancies": Object {},
  "d1Fellow": undefined,
  "getCertification": Object {
    "data": Object {},
    "loading": false,
  },
  "loading": undefined,
  "singleCertification": Object {},
}
`;

exports[`Test the certification container map state to props should return props value 1`] = `
Object {
  "certificationApplication": undefined,
  "certificationVacancies": Object {},
  "d1Fellow": undefined,
  "getCertification": Object {
    "data": Object {
      "1": Object {},
    },
    "loading": false,
  },
  "loading": undefined,
  "singleCertification": Object {},
}
`;
